doc/rm/otbn_coding_style.md

```diff
@@ -0,0 +1,232 @@
+---
+title: "OTBN Assembly Style Guide"
+---
+
+Where possible, OTBN assembly should follow the same principles as the [RISC-V assembly style guide]({{< relref "asm_coding_style" >}}).
+This guide describes additional OTBN-specific guidelines, and places where OTBN assembly can or should diverge from the RISC-V style guidelines.
+
+## General Advice
+
+### Register Names
+
+OTBN registers cannot be referred to by ABI names due to toolchain limitations.
+
+### Capitalization
+
+Lower case for mnemonics and registers.
+
+Example:
+```S
+  /* Correct */
+  bn.addi   w3, w3, 2
+
+  /* Wrong */
+  BN.ADDI   W3, W3, 2
+```
+
+Upper case for flags, flag groups, and half word identifiers in multiplication instructions.
+Example:
+```S
+  /* Correct */
+  bn.mulqacc.so  w27.L, w30.0, w25.1, 64
+  bn.sel    w3, w4, w2, FG1.C
+
+  /* Wrong */
+  bn.mulqacc.so  w27.l, w30.0, w25.1, 64
+  bn.sel    w3, w4, w2, fg1.c
+```
+
+### Pseudoinstrucions
+
+As in RISC-V, prefer pseudoinstructions (e.g. `ret`, `li`) in OTBN code where they exist.
+However, because `loop` and `loopi` instructions require counting the number of instructions in the loop body, it is permitted to avoid pseudoinstructions that can expand to multiple instructions within loop bodies.
+For example, `li` can expand to either 1 or 2 instructions depending on the immediate value.
+
+### Wrappers and ecall
+
+Generally, OTBN routines should be written so that they finish in `ret`.
+Use of `ecall` should be restricted to thin wrapper files which serve as an interface for the more substantial assembly files, usually just reading one or two arguments and then calling subroutines from the other files.
+
+### Comments
+
+Because of toolchain limitations, the `//` syntax is not available for OTBN.
```

Comment:
Clearly I'm bristling because I wrote the toolchain :-) But `otbn-as` and `riscv32-unknown-as` are the same in this respect. For `//` support in assembly, you have to pipe your code through `cpp`.

Reply:
I've rephrased to make it clear where the blame lies!